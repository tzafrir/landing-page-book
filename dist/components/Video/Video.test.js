'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _enzyme = require('enzyme');

var _enzymeAdapterReact = require('enzyme-adapter-react-16');

var _enzymeAdapterReact2 = _interopRequireDefault(_enzymeAdapterReact);

var _Video = require('./Video');

var _Video2 = _interopRequireDefault(_Video);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

(0, _enzyme.configure)({ adapter: new _enzymeAdapterReact2.default() });

describe('Video', function () {
  var props = void 0;
  var mounted = void 0;
  var video = function video() {
    if (!mounted) {
      mounted = (0, _enzyme.shallow)(_react2.default.createElement(_Video2.default, props));
    }
    return mounted;
  };
  describe('Video', function () {
    beforeEach(function () {
      props = {};
      mounted = undefined;
    });

    it('should render', function () {
      expect((0, _enzyme.shallow)(_react2.default.createElement(_Video2.default, props))).toMatchSnapshot();
    });

    it('always renders a div', function () {
      var divs = video().find('div');

      expect(divs.length).toBeGreaterThan(0);
    });
  });
});