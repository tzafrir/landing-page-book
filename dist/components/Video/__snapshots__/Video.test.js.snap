// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video Video should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Video
    className=""
    controls={true}
    loop={false}
    muted={false}
    onBuffer={[Function]}
    onDuration={[Function]}
    onEnded={[Function]}
    onError={[Function]}
    onPause={[Function]}
    onPlay={[Function]}
    onProgress={[Function]}
    onReady={[Function]}
    onSeek={[Function]}
    onStart={[Function]}
    playing={false}
    source=""
    volume={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ReactPlayer
        config={
          Object {
            "dailymotion": Object {
              "params": Object {
                "api": 1,
                "endscreen-enable": false,
              },
              "preload": false,
            },
            "facebook": Object {
              "appId": "1309697205772819",
            },
            "file": Object {
              "attributes": Object {},
              "forceAudio": false,
              "forceDASH": false,
              "forceHLS": false,
              "forceVideo": false,
              "hlsOptions": Object {},
              "tracks": Array [],
            },
            "mixcloud": Object {
              "options": Object {
                "hide_cover": 1,
              },
            },
            "soundcloud": Object {
              "options": Object {
                "buying": false,
                "download": false,
                "liking": false,
                "sharing": false,
                "show_comments": false,
                "show_playcount": false,
                "visual": true,
              },
            },
            "twitch": Object {
              "options": Object {},
            },
            "vimeo": Object {
              "playerOptions": Object {
                "autopause": false,
                "byline": false,
                "portrait": false,
                "title": false,
              },
              "preload": false,
            },
            "wistia": Object {
              "options": Object {},
            },
            "youtube": Object {
              "playerVars": Object {
                "iv_load_policy": 3,
                "modestbranding": 1,
                "playsinline": 1,
                "rel": 0,
                "showinfo": 0,
              },
              "preload": false,
            },
          }
        }
        controls={true}
        height="360px"
        loop={false}
        muted={false}
        onBuffer={[Function]}
        onDuration={[Function]}
        onEnded={[Function]}
        onError={[Function]}
        onPause={[Function]}
        onPlay={[Function]}
        onProgress={[Function]}
        onReady={[Function]}
        onSeek={[Function]}
        onStart={[Function]}
        playbackRate={1}
        playing={false}
        playsinline={false}
        progressInterval={1000}
        style={Object {}}
        url=""
        volume={1}
        width="640px"
        wrapper="div"
      />,
      "className": "Video ",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "config": Object {
          "dailymotion": Object {
            "params": Object {
              "api": 1,
              "endscreen-enable": false,
            },
            "preload": false,
          },
          "facebook": Object {
            "appId": "1309697205772819",
          },
          "file": Object {
            "attributes": Object {},
            "forceAudio": false,
            "forceDASH": false,
            "forceHLS": false,
            "forceVideo": false,
            "hlsOptions": Object {},
            "tracks": Array [],
          },
          "mixcloud": Object {
            "options": Object {
              "hide_cover": 1,
            },
          },
          "soundcloud": Object {
            "options": Object {
              "buying": false,
              "download": false,
              "liking": false,
              "sharing": false,
              "show_comments": false,
              "show_playcount": false,
              "visual": true,
            },
          },
          "twitch": Object {
            "options": Object {},
          },
          "vimeo": Object {
            "playerOptions": Object {
              "autopause": false,
              "byline": false,
              "portrait": false,
              "title": false,
            },
            "preload": false,
          },
          "wistia": Object {
            "options": Object {},
          },
          "youtube": Object {
            "playerVars": Object {
              "iv_load_policy": 3,
              "modestbranding": 1,
              "playsinline": 1,
              "rel": 0,
              "showinfo": 0,
            },
            "preload": false,
          },
        },
        "controls": true,
        "height": "360px",
        "loop": false,
        "muted": false,
        "onBuffer": [Function],
        "onDuration": [Function],
        "onEnded": [Function],
        "onError": [Function],
        "onPause": [Function],
        "onPlay": [Function],
        "onProgress": [Function],
        "onReady": [Function],
        "onSeek": [Function],
        "onStart": [Function],
        "playbackRate": 1,
        "playing": false,
        "playsinline": false,
        "progressInterval": 1000,
        "style": Object {},
        "url": "",
        "volume": 1,
        "width": "640px",
        "wrapper": "div",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ReactPlayer
          config={
            Object {
              "dailymotion": Object {
                "params": Object {
                  "api": 1,
                  "endscreen-enable": false,
                },
                "preload": false,
              },
              "facebook": Object {
                "appId": "1309697205772819",
              },
              "file": Object {
                "attributes": Object {},
                "forceAudio": false,
                "forceDASH": false,
                "forceHLS": false,
                "forceVideo": false,
                "hlsOptions": Object {},
                "tracks": Array [],
              },
              "mixcloud": Object {
                "options": Object {
                  "hide_cover": 1,
                },
              },
              "soundcloud": Object {
                "options": Object {
                  "buying": false,
                  "download": false,
                  "liking": false,
                  "sharing": false,
                  "show_comments": false,
                  "show_playcount": false,
                  "visual": true,
                },
              },
              "twitch": Object {
                "options": Object {},
              },
              "vimeo": Object {
                "playerOptions": Object {
                  "autopause": false,
                  "byline": false,
                  "portrait": false,
                  "title": false,
                },
                "preload": false,
              },
              "wistia": Object {
                "options": Object {},
              },
              "youtube": Object {
                "playerVars": Object {
                  "iv_load_policy": 3,
                  "modestbranding": 1,
                  "playsinline": 1,
                  "rel": 0,
                  "showinfo": 0,
                },
                "preload": false,
              },
            }
          }
          controls={true}
          height="360px"
          loop={false}
          muted={false}
          onBuffer={[Function]}
          onDuration={[Function]}
          onEnded={[Function]}
          onError={[Function]}
          onPause={[Function]}
          onPlay={[Function]}
          onProgress={[Function]}
          onReady={[Function]}
          onSeek={[Function]}
          onStart={[Function]}
          playbackRate={1}
          playing={false}
          playsinline={false}
          progressInterval={1000}
          style={Object {}}
          url=""
          volume={1}
          width="640px"
          wrapper="div"
        />,
        "className": "Video ",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "config": Object {
            "dailymotion": Object {
              "params": Object {
                "api": 1,
                "endscreen-enable": false,
              },
              "preload": false,
            },
            "facebook": Object {
              "appId": "1309697205772819",
            },
            "file": Object {
              "attributes": Object {},
              "forceAudio": false,
              "forceDASH": false,
              "forceHLS": false,
              "forceVideo": false,
              "hlsOptions": Object {},
              "tracks": Array [],
            },
            "mixcloud": Object {
              "options": Object {
                "hide_cover": 1,
              },
            },
            "soundcloud": Object {
              "options": Object {
                "buying": false,
                "download": false,
                "liking": false,
                "sharing": false,
                "show_comments": false,
                "show_playcount": false,
                "visual": true,
              },
            },
            "twitch": Object {
              "options": Object {},
            },
            "vimeo": Object {
              "playerOptions": Object {
                "autopause": false,
                "byline": false,
                "portrait": false,
                "title": false,
              },
              "preload": false,
            },
            "wistia": Object {
              "options": Object {},
            },
            "youtube": Object {
              "playerVars": Object {
                "iv_load_policy": 3,
                "modestbranding": 1,
                "playsinline": 1,
                "rel": 0,
                "showinfo": 0,
              },
              "preload": false,
            },
          },
          "controls": true,
          "height": "360px",
          "loop": false,
          "muted": false,
          "onBuffer": [Function],
          "onDuration": [Function],
          "onEnded": [Function],
          "onError": [Function],
          "onPause": [Function],
          "onPlay": [Function],
          "onProgress": [Function],
          "onReady": [Function],
          "onSeek": [Function],
          "onStart": [Function],
          "playbackRate": 1,
          "playing": false,
          "playsinline": false,
          "progressInterval": 1000,
          "style": Object {},
          "url": "",
          "volume": 1,
          "width": "640px",
          "wrapper": "div",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
